skopt/optimizer/base.py

```diff
@@ -209,8 +209,8 @@ def base_minimize(func, dimensions, base_estimator,
         For more details related to the OptimizeResult object, refer
         http://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.OptimizeResult.html
     """
-    specs = {"args": copy.copy(inspect.currentframe().f_locals),
-             "function": inspect.currentframe().f_code.co_name}
+    specs = {"args": copy.deepcopy(locals()),
```

Comment:
Given that there's no deepcopying (or copying even) in the other example below, is the deepcopy here necessary?

Have you checked the garbage graph afterwards? Are we safe?

Reply:
Good point I kept the copy to be safe but after visualizing the graph of `OptimizeResult` and I don't see any circular references. The obj graph is quite big because the `specs` contain the global `branin` function but that's expected. Worth noting that if you pass an object method as the func argument to `base_minimize`, that object will be referred to by `OptimizeResult` as long as `OptimizeResult` lives. 

![skopt_res_min](https://user-images.githubusercontent.com/41651716/117475459-12925800-af2a-11eb-9d10-0aeb8d3dd682.png)